nullaway/src/test/java/com/uber/nullaway/EnsuresNonNullTests.java

```diff
@@ -305,4 +301,84 @@ public void supportEnsuresAndRequiresNonNullContract() {
             "}")
         .doTest();
   }
+
+  @Test
+  public void ensuresNonNullWithStaticField() {
+    defaultCompilationHelper
+        .addSourceLines(
+            "Foo.java",
+            "package com.uber;",
+            "import javax.annotation.Nullable;",
+            "import com.uber.nullaway.annotations.EnsuresNonNull;",
+            "class Foo {",
+            "  @Nullable static Bar staticNullableItem;",
+            "  @EnsuresNonNull(\"staticNullableItem\")",
+            "  public void initializeStaticField() {",
+            "    staticNullableItem = new Bar();",
+            "  }",
+            "  public void useStaticField() {",
+            "    initializeStaticField();",
+            "    staticNullableItem.call();",
+            "  }",
+            "}")
+        .addSourceLines(
+            "Bar.java", "package com.uber;", "class Bar {", "  public void call() { }", "}")
+        .doTest();
+  }
+
+  @Test
+  public void requiresNonNullCallSiteErrorStaticField() {
+    defaultCompilationHelper
+        .addSourceLines(
+            "Foo.java",
+            "package com.uber;",
+            "import javax.annotation.Nullable;",
+            "import com.uber.nullaway.annotations.RequiresNonNull;",
+            "class Foo {",
+            "  static @Nullable Item staticNullableItem;",
+            "  ",
+            "  @RequiresNonNull({\"staticNullableItem\"})",
+            "  public void run() {",
+            "    staticNullableItem.call();",
+            "  }",
+            "  ",
+            "  public static void main(String[] args) {",
+            "    Foo foo = new Foo();",
+            "    // BUG: Diagnostic contains: Expected static field staticNullableItem to be non-null",
+            "    foo.run();",
+            "  }",
+            "}")
+        .addSourceLines(
+            "Item.java", "package com.uber;", "class Item {", "  public void call() { }", "}")
+        .doTest();
+  }
+
+  @Test
```

Comment:
Can we add a test combining `@EnsuresNonNull` and `@RequiresNonNull` for a static field?

Reply:
Added!